distributed/sizeof.py

```diff
@@ -39,29 +39,40 @@ def sizeof_numpy_ndarray(x):
 @sizeof.register_lazy("pandas")
 def register_pandas():
     import pandas as pd
+    import numpy as np
+
+    def object_size(x):
+        if not len(x):
+            return 0
+        sample = np.random.choice(x, size=20, replace=True)
+        sample = list(map(sizeof, sample))
+        return sum(sample) / 20 * len(x)
 
     @sizeof.register(pd.DataFrame)
     def sizeof_pandas_dataframe(df):
-        p = int(df.memory_usage(index=True).sum())
-        obj = int((df.dtypes == object).sum() * len(df) * 100)
-        if df.index.dtype == object:
-            obj += len(df) * 100
-        return int(p + obj) + 1000
+        p = sum(c.memory_usage(index=False) for _, c in df.iteritems())
```

Comment:
We removed this because it was too expensive.  We would like to perform this operation at well under a millisecond.